### PR TITLE
Support resubscribing a user to a single opt-in list.

### PR DESCRIPTION
**Pull request by jasonhildebrand**
Kudos to you for your work on this plugin.

I'm submitting this fix for an edge case I discovered in testing.  This is the scenario:
- the list is single opt-in
- a user was on the list previously, then unsubscribed
- now they wish to re-subscribe

In this case I was getting a "member compliance error" when they tried to re-subscribe.  The only way to get Mailchimp to re-subscribe them is to set their status to 'pending', which causes Mailchimp to send them a confirmation by email.

### CODE REVIEW

File: public/partials/shortcodes/process/process_form_submission.php [hunk 1]
```diff
@@ -177,8 +177,12 @@
 	// Check the opt-in value - is it double or single?
 	// Double opt-in means 'status_if_new' => 'pending'
 	$double_optin = isset( $optin_settings['optin'] ) ? (int) $optin_settings['optin'] : 0;
+        $was_unsubscribed = isset($member_exists['status']) && $member_exists['status'] == 'unsubscribed';
```
Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission.php]: Please use tabs for indentation instead of spaces.

Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission.php]: Include space inside the parentheses for `isset()`, like this:

```php
isset( $member_exists['status'] )
```

File: public/partials/shortcodes/process/process_form_submission.php [hunk 2]
```diff
@@ -177,8 +177,12 @@
...
 
-	if ( $double_optin === 1 ) {
+        // If the user was unsubscribed and is re-subscribing, we set the status to 'pending', which
+        // causes Mailchimp to send them a confirmation email.  This is the only way Mailchimp will
+        // allow us to re-subscribe the user.
```
Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission.php]: I would recommend moving this comment to just above where `$was_unsubscribed` is defined.

File: public/partials/shortcodes/process/process_form_submission_ajax.php [hunk 3]
```diff
@@ -140,8 +140,12 @@
 	// Check the opt-in value - is it double or single?
 	// Double opt-in means 'status_if_new' => 'pending'
 	$double_optin = isset( $optin_settings['optin'] ) ? (int) $optin_settings['optin'] : 0;
+        $was_unsubscribed = isset($member_exists['status']) && $member_exists['status'] == 'unsubscribed';
```
Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission_ajax.php]: The same comments from the above file apply to this one.

**Comment by jasonhildebrand** [comment]
OK, I've made the changes.

**Comment by kpu23** [comment]
Hi @jasonhildebrand,

Thanks for your interest and work with the plugin!

I think we can combine our work, in a way, to create a slightly simpler solution. 

We're currently setting the member's status to `pending` if the current member status is `pending` or `unsubscribed`. However, we're only doing it for double opt-in lists (see the variable `$double_optin_resubscribe`). Your code runs the same checks we're doing (i.e. `isset($member_exists['status']) && $member_exists['status'] == 'unsubscribed';`) but also includes this for single opt-ins.

If we change the current code to work for single opt-ins then it should achieve the same result. See my pull request here: https://github.com/yikesinc/yikes-inc-easy-mailchimp-extender/pull/729

Does that make sense? I think that's the way we should proceed; we don't need to check the member's status twice.

Your feedback is much appreciated.

Cheers,
Kevin.

**Comment by kpu23** [comment]
After some testing, there is a difference between our pull requests. In my pull request, I am checking for both the `unsubscribed` and `pending` statuses. In your pull request, you're only checking for the `unsubscribed` status. 

There is a certain niche situation where your check is better: 
- A user subscribes as double opt-in but never confirms - there status is `pending`
- The list is changed to single opt-in
- The same user subscribes again - with your code they will automatically be subscribed (status = `subscribed`). With my code, they will be sent a double opt-in email (status = `pending`).

**Pull request merged**